adoc/chapters/programming_interface.adoc

```diff
@@ -17640,40 +17635,500 @@ member function template.
+elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const DataT& rhs)
+----
+!====
+_Effects:_ Assigns the value [code]#rhs# to those elements of the underlying
+[code]#vec# object that have corresponding elements in the
+[code]#+__swizzled_vec__+# view.
+Elements in the underlying [code]#vec# object that do not have elements in the
+[code]#+__swizzled_vec__+# view are not assigned.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const vec<DataT, NumElements>& rhs)
+----
+!====
+_Effects:_ Assigns elements from [code]#rhs# to elements of the [code]#vec#
+object that underlies this [code]#+__swizzled_vec__+# view.
+The first element of [code]#rhs# is assigned to the element of the underlying
+[code]#vec# object that corresponds to the first element of the swizzle
+operation, etc.
+
+The order in which the elements are assigned is unspecified.
+Therefore, if the [code]#+__swizzled_vec__+# view has any repeated elements in
+the swizzle, those elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+|====
+
+[[table.functions.swizzled-vec]]
+.Hidden friend functions of the [code]#+__swizzled_vec__+# class template
+[separator="@"]
+|====
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template</*unspecified*/>                                                                 (1)
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs,
+                                          const __swizzled_vec__</*unspecified*/>& rhs)
+
+friend vec<DataT, NumElements> operatorOP(const vec<DataT, NumElements>& lhs,             (2)
+                                          const __swizzled_vec__& rhs);
+
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs,                    (3)
+                                          const vec<DataT, NumElements>& rhs)
+
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs, const DataT& rhs)  (4)
+friend vec<DataT, NumElements> operatorOP(const DataT& lhs, const __swizzled_vec__& rhs)  (5)
+----
+!====
+Where [code]#OP# is: [code]#pass:[+]#, [code]#-#, [code]#*#, [code]#/#,
+[code]#%#, [code]#&#, [code]#|#, [code]#^#, [code]#<<#, [code]#>>#.
+
+_Constraints:_ If [code]#OP# is one of the following: [code]#%#, [code]#&#,
+[code]#|#, [code]#^#, [code]#<<#, [code]#>>#; available only when: [code]#DataT
+!= float && DataT != double && DataT != half#.
+
+In addition, overload (1) is available only when the element data type of
+[code]#rhs# is the same as [code]#DataT# and when the number of elements in the
+[code]#rhs# view is equal to [code]#NumElements#.
+
+_Effects:_ These functions behave as though the swizzle operation represented
```

Comment:
I see two different wordings for describing this behavior, one describes it as performing the operation on the first element, then the second, etc, and the other describes this as evaluating into a temporary vec object and then performing the operator.

Could we use the same wording across all of these, and is it worth refactoring this wording into a separate section that we could reference from the individual member function/operator definitions?